javatests/arcs/core/host/PausableTestingHost.kt

```diff
@@ -0,0 +1,32 @@
+package arcs.core.host
+
+import arcs.core.storage.testutil.testStorageEndpointManager
+import arcs.jvm.util.testutil.FakeTime
+
+/** Fake [ArcHost] that let's us view pausing behavior for tests. */
+class PausableTestingHost(
+  schedulerProvider: SchedulerProvider,
+  vararg particles: ParticleRegistration
+) : TestingHost(
+  HandleManagerFactory(
+    schedulerProvider = schedulerProvider,
+    storageEndpointManager = testStorageEndpointManager(),
+    platformTime = FakeTime()
+  ), *particles
+) {
+
+  override suspend fun pause() {
+    super.pause()
+    numPauses++
+  }
+
+  override suspend fun unpause() {
+    super.unpause()
+    numUnpauses++
+  }
```

Comment:
If you go with the `NoopArcHost` idea above, you can get rid of the `super` calls, since they're just no-ops.

And if you go with the idea below of using an `isPaused` boolean, then you can also add in some safety checks, e.g.

```
override suspend fun pause() {
  assertThat(isPaused).isFalse()
  isPaused = true
}
```